ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/dataflow/reader_mcast_receiver_unary_sharded_ln_post_allgather.cpp

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include "dataflow_api.h"
+#include "hostdevcommon/common_values.hpp"
+
+
+// split REDUCE across cores
+void kernel_main() {
+
+    uint32_t reduce_sender_semaphore_addr    = get_semaphore(get_compile_time_arg_val(1));
+    constexpr uint32_t block_h                         = get_compile_time_arg_val(3);
+
+
+    constexpr uint32_t cb_ex_global = tt::CB::dataflow7; // [E[x], E[X^2]] global to all cores
+
+    #ifdef RMSNORM
+    constexpr uint32_t stats_tiles = 1;
+    #else
+    constexpr uint32_t stats_tiles = 2;
+    #endif
+
+    volatile tt_l1_ptr uint32_t* reduce_sender_semaphore_addr_ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(reduce_sender_semaphore_addr);
+
+    // inc mcast sender
+    noc_semaphore_set(reduce_sender_semaphore_addr_ptr, INVALID);
```

Comment:
Is there a way for us to ensure that the semaphore is initialized as INVALID from the host code so that we don't have to set it here? It is unlikely, but there could be a race condition if the sender sets this to valid before the receiver "initializes" it to invalid.

Reply:
@yugaoTT do you know if this is possible?

Reply:
I think the semaphores set to invalid during creation. The race condition is not possible though, I think the mcast sender waits for semaphore inc from receiver first, so sender will wait.